actions/global_actions.jsx

```diff
@@ -25,200 +26,192 @@ import {closeRightHandSide, closeMenu as closeRhsMenu, updateRhsState} from 'act
-        dialogProps: {
-            elements,
-        },
-    };
+    return (dispatch) => {
+        const submenuModalData = {
+            ModalId: ModalIdentifiers.MOBILE_SUBMENU,
+            dialogType: SubMenuModal,
+            dialogProps: {
+                elements,
+            },
+        };
 
-    dispatch(openModal(submenuModalData));
+        dispatch(openModal(submenuModalData));
+    };
 }
 
 export function sendEphemeralPost(message, channelId, parentId) {
-    const timestamp = Utils.getTimestamp();
-    const post = {
-        id: Utils.generateId(),
-        user_id: '0',
-        channel_id: channelId || getCurrentChannelId(getState()),
-        message,
-        type: PostTypes.EPHEMERAL,
-        create_at: timestamp,
-        update_at: timestamp,
-        root_id: parentId,
-        parent_id: parentId,
-        props: {},
+    return (dispatch, getState) => {
+        const timestamp = Utils.getTimestamp();
+        const post = {
+            id: Utils.generateId(),
+            user_id: '0',
+            channel_id: channelId || getCurrentChannelId(getState()),
+            message,
+            type: PostTypes.EPHEMERAL,
+            create_at: timestamp,
+            update_at: timestamp,
+            root_id: parentId,
+            parent_id: parentId,
+            props: {},
+        };
+
+        dispatch(handleNewPost(post));
     };
-
-    dispatch(handleNewPost(post));
 }
 
 export function sendAddToChannelEphemeralPost(user, addedUsername, addedUserId, channelId, postRootId = '', timestamp) {
-    const post = {
-        id: Utils.generateId(),
-        user_id: user.id,
-        channel_id: channelId || getCurrentChannelId(getState()),
-        message: '',
-        type: PostTypes.EPHEMERAL_ADD_TO_CHANNEL,
-        create_at: timestamp,
-        update_at: timestamp,
-        root_id: postRootId,
-        parent_id: postRootId,
-        props: {
-            username: user.username,
-            addedUsername,
-            addedUserId,
-        },
-    };
+    return (dispatch, getState) => {
+        const post = {
+            id: Utils.generateId(),
+            user_id: user.id,
+            channel_id: channelId || getCurrentChannelId(getState()),
+            message: '',
+            type: PostTypes.EPHEMERAL_ADD_TO_CHANNEL,
+            create_at: timestamp,
+            update_at: timestamp,
+            root_id: postRootId,
+            parent_id: postRootId,
+            props: {
+                username: user.username,
+                addedUsername,
+                addedUserId,
+            },
+        };
 
-    dispatch(handleNewPost(post));
+        dispatch(handleNewPost(post));
+    };
 }
 
 let lastTimeTypingSent = 0;
-export function emitLocalUserTypingEvent(channelId, parentPostId) {
-    const userTyping = async (actionDispatch, actionGetState) => {
-        const state = actionGetState();
+export function sendLocalUserTyping(channelId, parentPostId) {
```

Comment:
`send` is consistent with the other functions. This doesn't just trigger a redux action, it also includes sending a message on the `websocket`.

Reply:
we probably say `sendMessage` as action or `uploadFile` because `message` and `file` does not make a whole lot of sense. Here in this case `localUserTyping` in actions file pretty much explains what it supposed to do.  As i said i am 1/5 you can take a call about it.

Reply:
Makes sense. Will make this change.